src/actions.py

```diff
@@ -75,17 +75,18 @@ def optimize_limits_action(_action, _target, self):
     graphs.refresh(self)
 
 
-def restore_view_action(_action, _target, self):
-    self.canvas.load_limits()
-    graphs.refresh(self)
-
-
 def view_back_action(_action, _target, self):
-    self.canvas.dummy_toolbar.back()
+    if self.canvas.toolbar.back_enabled:
+        self.canvas.toolbar.back()
+        self.canvas.apply_limits()
+        # Fix weird view on back
+        self.canvas.queue_draw()
```

Comment:
It seems to improve, but it still sometimes gets stuck in shifted state. Using a regular `self.canvas.draw()` does not seem to have the issue.

Reply:
Okay, looking at the documentation, queue_draw waits on the GUI, so when its called multiple times like draw_idle, it only renders once. However since the `back()` function calls draw_idle down the line, that is maybe the root problem here.